src/main/java/com/conveyal/analysis/models/AnalysisRequest.java

```diff
@@ -178,55 +162,34 @@
      * TODO arguably this should be done by a method on the task classes themselves, with common parts factored out
      *      to the same method on the superclass.
      */
-    public AnalysisWorkerTask populateTask (AnalysisWorkerTask task, Project project) {
-
-        // Fetch the modifications associated with this project, filtering for the selected scenario
-        // (denoted here as "variant"). There are no modifications in the baseline scenario
-        // (which is denoted by special index -1).
-        List<Modification> modifications = new ArrayList<>();
-        String scenarioName;
-        if (variantIndex > -1) {
-            if (variantIndex >= project.variants.length) {
-                throw AnalysisServerException.badRequest("Scenario does not exist. Please select a new scenario.");
-            }
-            modifications = modificationsForProject(project.accessGroup, projectId, variantIndex);
-            scenarioName = project.variants[variantIndex];
-        } else {
-            scenarioName = "Baseline";
-        }
+    public AnalysisWorkerTask populateTask (AnalysisWorkerTask task, List<Modification> modifications) {
+        if (bounds == null) throw AnalysisServerException.badRequest("Analysis bounds must be set.");
 
-        // The CRC of the modifications in this scenario is appended to the scenario ID to
-        // identify a unique revision of the scenario (still denoted here as variant) allowing
-        // the worker to cache and reuse networks built by applying that exact revision of the
+        // The CRC of the modifications in this scenario is appended to the bundle ID to identify a unique set of
+        // modifications allowing the worker to cache and reuse networks built by applying that exact revision of the
         // scenario to a base network.
         CRC32 crc = new CRC32();
         crc.update(JsonUtilities.objectToJsonBytes(modifications));
         long crcValue = crc.getValue();
-
-        task.scenario = new Scenario();
         // FIXME Job IDs need to be unique. Why are we setting this to the project and variant?
         //       This only works because the job ID is overwritten when the job is enqueued.
         //       Its main effect is to cause the scenario ID to have this same pattern!
         //       We should probably leave the JobID null on single point tasks. Needed: polymorphic task initialization.
-        task.jobId = String.format("%s-%s-%s", projectId, variantIndex, crcValue);
-        task.scenario.id = task.scenarioId = task.jobId;
-        task.scenario.modifications = modifications;
-        task.scenario.description = scenarioName;
-        task.graphId = project.bundleId;
+        task.jobId = String.format("%s-%s", bundleId, crcValue);
```

Comment:
While we're changing this, we should fix the strangeness mentioned in the FIXME above about jobIds. I think jobId can be nulled out here, and in subsequent lines we can directly set `scenario.id = task.scenarioId = String.format("%s-%s", this.scenarioId, crcValue);` In the PR code the first part of the scenario name is the bundleId. Is this intentional? Should it instead be the scenarioId?

The code in this PR has the first part of the generated (CRC-versioned) scenarioId as the bundleId instead of scenarioId, is this intentional?